src/snowcli/cli/nativeapp/manager.py

```diff
@@ -85,30 +125,85 @@ def artifacts(self) -> List[ArtifactMapping]:
     def deploy_root(self) -> Path:
         return Path(self.project_root, self.definition["deploy_root"])
 
+    @cached_property
+    def package_scripts(self) -> List[str]:
+        """
+        Relative paths to package scripts from the project root.
+        """
+        return self.definition.get("package", {}).get("scripts", [])
+
+    @cached_property
+    def stage_fqn(self) -> str:
+        return f'{self.package_name}.{self.definition["source_stage"]}'
+
+    @cached_property
+    def stage_schema(self) -> Optional[str]:
+        return extract_schema(self.stage_fqn)
+
+    @cached_property
+    def warehouse(self) -> Optional[str]:
+        return self.definition.get("application", {}).get("warehouse", None)
+
+    @cached_property
+    def project_identifier(self) -> str:
+        return clean_identifier(self.definition["name"])
+
+    @cached_property
+    def package_name(self) -> str:
+        return self.definition.get("package", {}).get(
+            "name", default_app_package(self.project_identifier)
+        )
+
+    @cached_property
+    def package_role(self) -> str:
+        return self.definition.get("package", {}).get("role", None) or default_role()
+
+    @cached_property
+    def app_name(self) -> str:
+        return self.definition.get("application", {}).get(
+            "name", default_application(self.project_identifier)
+        )
+
+    @cached_property
+    def app_role(self) -> str:
+        return (
+            self.definition.get("application", {}).get("role", None) or default_role()
+        )
+
+    @cached_property
+    def debug_mode(self) -> bool:
+        return self.definition.get("application", {}).get("debug", True)
```

Comment:
Should we have a generic method for this? Like `self.definition.get("application", "debug", default=True)

Reply:
I think it goes hand-in-hand with the DTO question you had above. Putting these methods closer to `project/schema.py` would be a win for discovery.